src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -48,47 +48,65 @@ internal static void SetSslOptions(EasyRequest easy, ClientCertificateOption cli
                 // Register the callback with libcurl.  We need to register even if there's no user-provided
                 // server callback and even if there are no client certificates, because we support verifying
                 // server certificates against more than those known to OpenSSL.
-                CURLcode answer = easy.SetSslCtxCallback(s_sslCtxCallback, userPointer);
-                switch (answer)
+                if (!CurlSslVersionDescription.StartsWith("openssl", StringComparison.OrdinalIgnoreCase) ||
+                    CurlSslVersionDescription.IndexOf("openssl/1.0", StringComparison.OrdinalIgnoreCase) != -1)
```

Comment:
Yeah, the fact that it would have worked for wolfSSL was probably just a bug waiting to happen (since we'd take the pointers from wolfSSL and send them off to OpenSSL for interpretation).  https://curl.haxx.se/libcurl/c/CURLOPT_SSL_CTX_FUNCTION.html.